libusb/core.c

```diff
@@ -2313,12 +2313,8 @@ int API_EXPORTED libusb_init(libusb_context **ctx)
 	list_init(&_ctx->usb_devs);
 	list_init(&_ctx->open_devs);
 
-	/* default context should be initialized before calling usbi_dbg */
-	if (!ctx) {
-		usbi_default_context = _ctx;
-		default_context_refcnt = 1;
-		usbi_dbg(usbi_default_context, "created default context");
-
+	if (ctx || !usbi_default_context) {
```

Comment:
i've been trying to wrap my head around this a little more and i believe this condition is always true and can be removed for clarity.

the failing case of `!ctx && usbi_default_context` is handled on line 2282 above: https://github.com/libusb/libusb/pull/942/files#diff-c1f9bc250077d41456a3e580fca0ddf5d8c25b741bff6d9b9505990a8b70b254R2282

the present code appears to forward all default options to backends in libusb_init .

Reply:
Yes, I think you are right, the expression will always be true. Yes, it forwards all default options to the context being created, except for LIBUSB_OPTION_LOG_LEVEL. Are there others that shouldn't be?

Reply:
eh I'm trying to figure out how best to shoehorn my pr in, that's all.  it takes a pointer argument.
it looks fine for what's in the tree, but I found the new code clearer with the always-true test removed.

Reply:
Alright. Yes, we should remove the always-true test, it ended up there for "historical reasons" I guess.